client/js/table-view.js

```diff
@@ -62,13 +63,52 @@ class TableView {
       }
       docFragment.appendChild(tr);
     }
-    removeChildren(this.bodyEle);
-    this.bodyEle.appendChild(docFragment);
+    removeChildren(this.bodyEl);
+    this.bodyEl.appendChild(docFragment);
+  }
+
+  renderTableFoot() {
+    removeChildren(this.footEl);
+    const docFragment = document.createDocumentFragment();
+    const tr = createTR();
+    for(let col = 0; col < this.model.numOfCols; col++) {
+      const colVals = this.fetchColumnVals.bind(this, col);
+      let sum = '';
+      if (colVals().length > 0) {
+        sum = this.computeColumnSum(colVals()); 
+      }
+      const td = createTD(sum.toString());
+      tr.appendChild(td);
+    }
+    docFragment.appendChild(tr);
+    this.footEl.appendChild(docFragment);
+    this.setFootColor();
+  }
+
+  fetchColumnVals(colIndex) {
+    return getTableRows()
+      .map((tr) => {
+        const cellVal = parseInt(tr.cells[colIndex].textContent, 10); 
+        return this.validateColumnVals(cellVal);
+      })
+      .filter((cellVal) => cellVal !== '');
+  }
+
+  validateColumnVals(cellVal) {
+    return (!Number.isNaN(cellVal)) ? cellVal : '';
+  }
+
+  computeColumnSum(colVals) {
+    return colVals.reduce((sum, cellVal) => sum + cellVal);
+  }
+
+  setFootColor() {
```

Comment:
Furthermore I don't think there's much benefit to factoring out this helper function. It may as well go back inline with the rest of the DOM manipulation occurring in the calling function.

Why? Because 

* it's only called in one place, 
* the body is only one line long, 
* that one line is doing a trivial amount of processing, 
* that line is completely self-describing as it is.

We don't gain any re-use or "easier to think about because the details are abstracted away" benefits from this particular helper function so I propose you just discard it.